SwiftCompilerSources/Sources/AST/Declarations.swift

```diff
@@ -29,24 +29,62 @@ public class Decl: CustomStringConvertible, Hashable {
   // True if this declaration is imported from C/C++/ObjC.
   public var hasClangNode: Bool { bridged.hasClangNode() }
 
+  public var declContext: DeclContext {
+    if let decl = parent {
+      return decl as! DeclContext
+    }
+    return DeclContextObj(bridged: bridged.getDeclContext())
+  }
+
+  public var bridgedDecl: BridgedDecl { BridgedDecl(raw: bridged.obj) }
+
   public static func ==(lhs: Decl, rhs: Decl) -> Bool { lhs === rhs }
 
   public func hash(into hasher: inout Hasher) {
     hasher.combine(ObjectIdentifier(self))
   }
+
+  public func setImplicit() { bridged.setImplicit() }
+}
+
```

Comment:
I would prefer to put `DeclContext` into its own sourcefile. Even if the file will be very small - initially